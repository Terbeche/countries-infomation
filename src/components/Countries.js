import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchCountries } from '../redux/countries/countries';
import CountriesList from './countriesComponents/CountriesList';

const Countries = () => {
  const dispatch = useDispatch();
  const { loading, Countries } = useSelector((state) => state.countries);

  useEffect(() => {
    if (Countries.length === 0) {
      dispatch(fetchCountries());
    }
  }, [dispatch]);

  return (

    <div>
      {loading && <div>Loading ...</div>}
      {!loading && (<CountriesList />)}

    </div>
  );
};

export default Countries;
