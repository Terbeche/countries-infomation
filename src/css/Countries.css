.country{
  display: flex;
  margin-bottom: 2rem;
}

.country-detail{
  display: flex;
  flex-direction: column;
}

.name {
  margin-bottom: 1rem;
}

.population {
  margin-bottom: 1rem;
  line-height: 1.5rem;
}

.countries-container {
  margin-top: 10rem;
}
