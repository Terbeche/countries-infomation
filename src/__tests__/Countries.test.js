import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Countries from '../components/Countries';

it('Countries component snapshot testing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Countries />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
