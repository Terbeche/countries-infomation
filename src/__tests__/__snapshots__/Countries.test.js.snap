// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries component snapshot testing 1`] = `
<div>
  <div
    className="countries-container"
  />
</div>
`;
